### DISCUSSION TITLE
GPG signatures in non-Maven-P2 repositories

### DISCUSSION THREAD

**Discussion by reckart** [eclipse-tycho/tycho#946] [Q&A]
The [tycho-p2-repository:assemble-maven-repository](https://www.eclipse.org/tycho/sitedocs/tycho-p2/tycho-p2-repository-plugin/assemble-maven-repository-mojo.html) mojo has an option `includePGPSignature` which embeds the PGP signatures of the Maven artifacts in the P2 repository metadata.

Unfortunately, the [tycho-p2-repository:assemble-repository](https://www.eclipse.org/tycho/sitedocs/tycho-p2/tycho-p2-repository-plugin/assemble-repository-mojo.html) mojo does not have this option.

Is `includePGPSignature` something that could be added to `tycho-p2-repository:assemble-repository`, or does it not make sense?

**Comment by laeubi** [comment]
> Is `includePGPSignature` something that could be added to `tycho-p2-repository:assemble-repository`, or does it not make sense?

It is (not yet) supported see for example https://github.com/eclipse/tycho/issues/661 but you can (re) (pgp) sign it with the https://www.eclipse.org/tycho/sitedocs/tycho-gpg-plugin/sign-p2-artifacts-mojo.html

**Reply by reckart** [reply]
Thanks for the quick response. That plugin kind-of works. It generates the `.asc` files and I can also see the `gpg.*` properties appear in the `artifacts.xml` - but for some reason the `pgp.publicKeys` is **HUGE** - and the `artifacts.xml` gets blown up to like 30MB. I have decoded what it puts there, and it is **ALL** public keys from my keyring! The `includePGPSignature` from `[tycho-p2-repository:assemble-maven-repository](https://www.eclipse.org/tycho/sitedocs/tycho-p2/tycho-p2-repository-plugin/assemble-maven-repository-mojo.html)` only embeds the public key of my default identity.

**Reply by laeubi** [reply]
This is probably a bug you can report here https://github.com/eclipse/tycho/issues
The `assemble-maven-repository` works quite different, it reuses already attached signatures and fetches the public key from the key-server so it is independent from your local configuration (but requires you have published the key already).

**Reply by reckart** [reply]
==> https://github.com/eclipse/tycho/issues/948

**Reply by reckart** [reply]
Apparently the PGP signatures included in the p2 repositories are not supported yet by Eclipse. 

There is an [old thread](https://www.eclipse.org/lists/p2-dev/msg05910.html) from 2021 regarding adding support for PGP signatures when validating bundles during install, but I tried Eclipse 2022-03 and Eclipse 2022-06-R1 with the repos signed by the 
`tycho-gpg-plugin`, but both do not recognizes the signatures.

<img width="914" alt="Screenshot 2022-05-06 at 14 53 48" src="https://user-images.githubusercontent.com/1410238/167135255-97eb9b09-7d11-4331-9d6c-fb0c7a804740.png">

**Reply by reckart** [reply]
@mickaelistria has there been any news on the PGP support in Eclipse since that [thread](https://www.eclipse.org/lists/p2-dev/msg05910.html) you started in 2021?

**Reply by mickaelistria** [reply]
PGP signatures are supported. Please verify that the artifacts you're trying to install do actually have signature metadata, and that there isn't another repository providing an unsigned version of the artifacts during installation.

**Reply by reckart** [reply]
I've been able to get Eclipse to recognize the PGP signatures in a minimal example... now I need to figure out why it doesn't work in the actual build that I'm working on... 🤔

**Reply by laeubi** [reply]
@mickaelistria can you take a look into this issue?

- https://github.com/eclipse/tycho/issues/948
